### ISSUE TITLE
Errors seen in the nginx controller logs `Failed to update lock: configmaps "ingress-controller-leader" is forbidden` 

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#35583]
**Rancher Server Setup**
- Rancher version: `v2.6-head 96e2438 `
- Installation option (Docker install/Helm Chart): Docker install
   - If Helm Chart, Kubernetes Info: k3s
         Cluster Type (RKE1, RKE2, k3s, EKS, etc): 
         Node Setup: 1 node 
         Version: `v1.22.3+k3s1`
- Proxy/Cert Details: `self-signed`

**Information about the Cluster**
- Kubernetes version: `v1.22.3`
- Cluster Type (Local/Downstream): `Downstream custom 2 worker, 1 etcd, 1 cp` RKE1 custom cluster

**Describe the bug**
Error logs are seen flowing continuously in the nginx-controller pod logs in the downstream clusters created on k8s v1.22.3

**To Reproduce**
Create a downstream cluster on k8s v1.22.3
From the explore cluster --> nginx-controller daemon set in the namespace `ingress-nginx`  verify the logs on the pods

**Result**
Following logs are seen continuously for every few seconds:
```
E1117 23:44:55.988115 7 leaderelection.go:361] Failed to update lock: configmaps "ingress-controller-leader" is forbidden: User "system:serviceaccount:ingress-nginx:ingress-nginx" cannot update resource "configmaps" in API group "" in the namespace "ingress-nginx"
Wed, Nov 17 2021 3:45:09 pm | E1117 23:45:09.500147 7 leaderelection.go:361] Failed to update lock: configmaps "ingress-controller-leader" is forbidden: User "system:serviceaccount:ingress-nginx:ingress-nginx" cannot update resource "configmaps" in API group "" in the namespace "ingress-nginx"
Wed, Nov 17 2021 3:45:18 pm | E1117 23:45:18.416296 7 leaderelection.go:361] Failed to update lock: configmaps "ingress-controller-leader" is forbidden: User "system:serviceaccount:ingress-nginx:ingress-nginx" cannot update resource "configmaps" in API group "" in the namespace "ingress-nginx"
Wed, Nov 17 2021 3:45:30 pm | E1117 23:45:30.101453 7 leaderelection.go:361] Failed to update lock: configmaps "ingress-controller-leader" is forbidden: User "system:serviceaccount:ingress-nginx:ingress-nginx" cannot update resource "configmaps" in API group "" in the namespace "ingress-nginx"

```

**Expected Result**
Expected no error logs in the nginx controller pods.

**Comment by kinarashah** [comment]
Fix merged in RKE1, available to test after is merged in Rancher `v2.6.3-rc1`.

**Comment by brendarearden** [comment]
Available to test on v2.6.3-rc1

**Comment by anupama2501** [comment]
Verified on v2.6-head 4375161

Created a cluster on v1.22.3-rancher1-1 and verified no error logs mentioned in the ticket are seen from the nginx-controller
